endpointmanager/pkg/capabilityhandler/capabilityreceiver_integration_test.go

```diff
@@ -0,0 +1,194 @@
+// +build integration
+
+package capabilityhandler
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/config"
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/endpointmanager"
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/endpointmanager/postgresql"
+	th "github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/testhelper"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+var store *postgresql.Store
+
+var hitps []*endpointmanager.HealthITProduct = []*endpointmanager.HealthITProduct{
+	&endpointmanager.HealthITProduct{
+		Name:                 "EpicCare Ambulatory Base",
+		Version:              "February 2020",
+		Developer:            "Epic Systems Corporation",
+		CertificationStatus:  "Active",
+		CertificationDate:    time.Date(2020, 2, 20, 0, 0, 0, 0, time.UTC),
+		CertificationEdition: "2015",
+		CHPLID:               "15.04.04.1447.Epic.AM.13.1.200220",
+		APIURL:               "https://open.epic.com/Interface/FHIR",
+	},
+	&endpointmanager.HealthITProduct{
+		Name:                 "PowerChart (Clinical)",
+		Version:              "2018.01",
+		Developer:            "Cerner Corporation",
+		CertificationStatus:  "Active",
+		CertificationDate:    time.Date(2018, 7, 27, 0, 0, 0, 0, time.UTC),
+		CertificationEdition: "2015",
+		CHPLID:               "15.04.04.1221.Powe.18.03.1.180727",
+		APIURL:               "http://fhir.cerner.com/authorization/",
+	},
+	&endpointmanager.HealthITProduct{
+		Name:                 "Health Services Analytics",
+		Version:              "8.00 SP1-SP5",
+		Developer:            "Cerner Health Services, Inc.",
+		CertificationStatus:  "Withdrawn by Developer",
+		CertificationDate:    time.Date(2017, 12, 5, 0, 0, 0, 0, time.UTC),
+		CertificationEdition: "2014",
+		CHPLID:               "14.07.07.1222.HEA5.03.01.1.171205",
+	},
+}
+
+func TestMain(m *testing.M) {
+	var err error
+
+	err = config.SetupConfigForTests()
+	if err != nil {
+		panic(err)
+	}
+
+	err = setup()
+	if err != nil {
+		panic(err)
+	}
+
+	hap := th.HostAndPort{Host: viper.GetString("dbhost"), Port: viper.GetString("dbport")}
+	err = th.CheckResources(hap)
+	if err != nil {
+		panic(err)
+	}
+
+	code := m.Run()
+
+	teardown()
+	os.Exit(code)
+}
+
+func Test_saveMsgInDB(t *testing.T) {
+	teardown, _ := th.IntegrationDBTestSetup(t, store.DB)
```

Comment:
It looks like here as well as in productquerier_integration_test, populatefhirendpoints_integration_test, and capabilityinfo_integration_test you could just use the new th.IntegrationDBTestSetupMain in the TestMain for those files, unless there's something I'm missing

Reply:
the setup ensure the db is empty. the teardown clears out the db.
in the integration tests, we want a clean db for each individual test.
however, the way the e2e test is written, it depends on data that was previously entered in the db. so the integration tests do it on a test-by-test basis, and the e2e does it once in main.

Reply:
oh okay makes sense!